docs/language/blocks-and-lambdas.mdx

```diff
@@ -495,3 +602,42 @@ class int:
 ```
 
 Note: non-named block parameters have to be the last in the function signature.
+
+Similarly, you can define a function that takes a lambda as an argument:
+
+```
+call-lambda callback/Lambda:
+  callback.call 42
+
+main:
+  call-lambda:: print it
+```
+
+A lambda is an object of type `Lambda`. You can store it in a variable, pass it
+as an argument, and call it later. The `call` method is used to execute the
+lambda.
+
+### Lambdas vs. blocks: choosing the right tool
+
+Only the receiver of a callback can determine whether to use a block or a lambda. As
+a user, one has to follow the type declared in the method signature.
+
+When writing functions that accept callbacks, consider the following guidelines:
+- Use blocks for short-lived, context-dependent callbacks that don't need to be
+  stored or returned. Blocks are significantly more efficient and should be the
+  default choice when possible.
+- Use lambdas when you need to store the function.
+- Consider providing both block and lambda versions if you need to support both
+  use cases. The lambda version can always redirect to the block version:
+  ```
+  my-function callback:
```

Comment:
Maybe flip the order of these two functions?

Reply:
done.